src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -403,7 +403,10 @@ protected void CheckDocumentIsOpen(DocumentId documentId)
 
                 // Fire and forget that the workspace is changing.
                 RaiseWorkspaceChangedEventAsync(WorkspaceChangeKind.DocumentChanged, oldSolution, newSolution, documentId: documentId);
+
+                // We fire 2 events on source document opened.
                 this.RaiseDocumentOpenedEventAsync(newDoc);
+                this.RaiseTextDocumentOpenedEventAsync(newDoc);
```

Comment:
Should this call just be moved into the other one, so we can't ever forget to fire both?

Reply:
I thought about it, but then we already have one more explicit `WorkspaceChangedEvent` being fired just above this, so felt it is better to keep all event raising explicit for consistency.